examples/threathunter/threat_intelligence/stix_parse.py

```diff
@@ -0,0 +1,348 @@
+                                'description': description,
+                                'title': title,
+                                'timestamp': timestamp,
+                                'link': link,
+                                'score': score})
+
+    elif type(props) == Address:
+        if props.category == 'ipv4-addr' and props.address_value:
+            iocs = {'ipv4': []}
+
+            #
+            # Sometimes props.address_value.value is a list vs a string
+            #
+            if type(props.address_value.value) is list:
+                for ip in props.address_value.value:
+                    if validate_ip_address(ip.strip()):
+                        iocs['ipv4'].append(ip.strip())
+            else:
+                ipv4 = props.address_value.value.strip()
+                if validate_ip_address(ipv4):
+                    iocs['ipv4'].append(ipv4)
+
+            if len(iocs['ipv4']) > 0:
+                reports.append({'iocs_v2': iocs,
+                                'id': sanitize_id(observable.id_),
+                                'description': description,
+                                'title': title,
+                                'timestamp': timestamp,
+                                'link': link,
+                                'score': score})
+
+    elif type(props) == File:
+        iocs = {'md5': []}
+        if props.md5:
+            if type(props.md5) is list:
+                for md5 in props.md5:
+                    if validate_md5sum(md5.strip()):
+                        iocs['md5'].append(md5.strip())
+            else:
+                if hasattr(props.md5, 'value'):
+                    md5 = props.md5.value.strip()
+                else:
+                    md5 = props.md5.strip()
+                if validate_md5sum(md5):
+                    iocs['md5'].append(md5)
+
+            if len(iocs['md5']) > 0:
+                reports.append({'iocs_v2': iocs,
+                                'id': sanitize_id(observable.id_),
+                                'description': description,
+                                'title': title,
+                                'timestamp': timestamp,
+                                'link': link,
+                                'score': score})
+
+    logger.debug(f"sending reports: {reports}")
+
+    return reports
+
+
+def get_stix_indicator_score(indicator, default_score):
+    if not indicator.confidence:
+        return default_score
+
+    confidence_val_str = str(indicator.confidence.value)
+    if confidence_val_str.isdigit():
+        score = int(indicator.confidence.to_dict().get("value", default_score))
+        return score
+    if confidence_val_str.lower() == "high":
+        return 7  # 75
+    if confidence_val_str.lower() == "medium":
+        return 5  # 50
+    if confidence_val_str.lower() == "low":
+        return 2  # 25
+
+    return default_score
+
+
+def get_stix_indicator_timestamp(indicator):
```

Comment:
Make sure we are using ISO8601 format https://www.iso.org/iso-8601-date-and-time-format.html

Here is an example of the parser being added to cbapi https://github.com/carbonblack/cbapi-python/pull/198/files#diff-cf21c9af82582239fe2ef85c18694d3fR42

Reply:
Referencing your first comment, I didn't make a change to the timestamps because I'm unsure if it makes more sense for the user to have STIX timestamps, or timestamps from when the report was created

Reply:
After consideration, I'll keep the timestamp data from the TAXII server. Users could find this information useful as a point of reference for when a threat was discovered and added to their ThreatIntel source